modules/HapticGlove_module/src/GloveControlHelper.cpp

```diff
@@ -201,6 +202,22 @@ bool GloveControlHelper::setFingertipVibrotactileFeedbackReferences(
 
     return m_pImp->setFingertipVibrotactileValues(m_desiredVibrotactileValues);
 }
+
+bool GloveControlHelper::setFingertipHapticFeedbackReferences()
+{
+
+    for (size_t i = 0; i < m_numForceFeedback; i++)
+    {
+        m_desiredHapticValues[i] = m_desiredForceValues[i];
+    }
+    for (size_t i = 0; i < m_numVibrotactileFeedback; i++)
+    {
+        m_desiredHapticValues[i + m_numForceFeedback] = m_desiredVibrotactileValues[i];
```

Comment:
Would it be possible to use separate vectors?

Reply:
It is possible. But can you please explain more on why do you think it is better?

Reply:
It is better to separate different data. As it is right now the receiver also needs to know that half of it is force, the other half is haptic. It is very error-prone, also because you might have a different number of haptic and force feedback values

Reply:
I tried to separate the vectors with this commit: https://github.com/robotology/walking-teleoperation/pull/139/commits/5c692d3f3db0904230bef887230277d846a7ebaa

_Note: will need to test it though!_